app/src/main/java/com/concordium/wallet/data/preferences/Preferences.kt

```diff
@@ -15,9 +19,145 @@ open class Preferences {
+        if(authPreferences.getBoolean(SHARED_PREFERENCES_ARE_ENCRYPTED, false)){
+            sharedPreferences = initializeEncryptedSharedPreferences(context, preferenceName)
+        }else{
+            sharedPreferences =
+                if(migratePreferencesSuccess(context, preferenceName, preferenceMode, authPreferences)){
+                    initializeEncryptedSharedPreferences(context, preferenceName)
+                }else{
+                    getSharedPreferences(context, preferenceName, preferenceMode)
+                }
+        }
+    }
+
+    /**
+     * Returns the required SharedPreferences
+     * @return [SharedPreferences] *if* unencrypted data is still present
+     *
+     * *else* an instance of encrypted [SharedPreferences]
+     */
+    private fun getSharedPreferences(
+        context: Context,
+        preferenceName: String,
+        preferenceMode: Int
+    ): SharedPreferences {
+        val unEncryptedSharedPreferences = context.getSharedPreferences(preferenceName, preferenceMode)
+
+        return if(unEncryptedSharedPreferences.all.isEmpty()) {
+            initializeEncryptedSharedPreferences(context, preferenceName)
+        }else{
+            unEncryptedSharedPreferences
+        }
+    }
+
+    /**
+     * Loops through all of the [SharedPreferences] and attempts to save them to [EncryptedSharedPreferences]
+     * @return *true* if the [SharedPreferences] don't need migrating
+     *
+     * *true* if [SharedPreferences] is successfully migrated and the old data cleared
+     *
+     * *false* if the migrations fail or is not cleared successfully
+     */
+    private fun migratePreferencesSuccess(
+        mContext: Context,
+        preferenceName: String,
+        preferenceMode: Int,
+        authPreferences: SharedPreferences): Boolean{
+        var allPreferencesAreMigrated = true
+        var continueWithEncryptedSharedPreference = true
+
+        for(prefName in SharedPreferencesKeys.PREF_ALL){
+            if(!migrateSinglePreferencesIfNeededOrContinue(mContext, prefName, preferenceMode)){
+                allPreferencesAreMigrated = false
+
+                if(prefName == preferenceName) {
+                    continueWithEncryptedSharedPreference = false
+                }
+            }
+        }
+        if(allPreferencesAreMigrated){
+            authPreferences.edit().putBoolean(SHARED_PREFERENCES_ARE_ENCRYPTED, true).commit()
+        }
+        return  continueWithEncryptedSharedPreference
+    }
+
+    /**
+     * Migrate single [SharedPreferences] to [EncryptedSharedPreferences]
+     * @return *true* if the unencrypted preference is empty
+     *
+     * *true* if the preference is successfully migrated and cleared
+     *
+     * *false* if the migration fails
+     *
+     * *false* if the old preference in not cleared
+     */
+    private fun migrateSinglePreferencesIfNeededOrContinue(
```

Comment:
Why is it named "or continue"?

Reply:
The code loops through all of the SharedPreferences and checks if a migration is needed. If a migration is not needed then it continues to the next one.

Reply:
Ok I will rename it to migrateSinglePreferences